src/components/RegisterApollo.js

```diff
@@ -0,0 +1,187 @@
+import React, { useEffect, useState } from "react";
+import {
+  Button,
+  Container,
+  FormControl,
+  FormGroup,
+  FormHelperText,
+  Input,
+  InputLabel,
+  Typography,
+  makeStyles,
+} from "@material-ui/core";
+import { Redirect } from "react-router-dom";
+import { useMutation } from "@apollo/react-hooks";
+import { LOGIN } from "../gql/login";
+import { REGISTER } from "../gql/register";
+import { withAuthConfigApollo } from "./withAuthConfigApollo";
+
+const useStyles = makeStyles({
+  container: {
+    margin: "3vh auto",
+  },
+  disclaimer: {
+    margin: "1vmax",
+  },
+  form: {
+    margin: "2vh auto",
+  },
+});
+
+const Register = ({ setGoogleUser, location, history }) => {
+  const classes = useStyles();
+  const [form, setForm] = useState({});
+  const [register, { data: registerData, error: registerError }] = useMutation(
+    REGISTER
+  );
+  const [login, { data: loginData, error: loginError }] = useMutation(LOGIN);
+
+  useEffect(() => {
+    if (registerError || loginError) {
+      console.log("register error:", registerError);
+      console.log("login error:", loginError);
+      history.replace({ location: "/" });
+    }
```

Comment:
I think you can move this outside useEffect and it should have the same behaviour